.github/workflows/eden_rpi.yml

```diff
@@ -0,0 +1,140 @@
+  workflow_run:
+    workflows:
+      - Publish
+    types:
+      - completed
+# yamllint disable rule:line-length
+jobs:
+  integration:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        hv: ["kvm", "xen"]
+    steps:
+      - name: Check secrets
+        run: |
+          if [ -z "$ROL_API_URL" ]; then echo "::error::ROL_API_URL is empty" && exit 1; fi
+          if [ -z "$ROL_API_KEY" ]; then echo "::error::ROL_API_KEY is empty" && exit 1; fi
+          if [ -z "$ROL_PROJECT" ]; then echo "::error::ROL_PROJECT is empty" && exit 1; fi
+          if [ -z "$ROL_OVPN_CONF_BASE64" ]; then echo "::error::ROL_OVPN_CONF_BASE64 is empty" && exit 1; fi
+        env:
+          ROL_API_URL: ${{ secrets.ROL_API_URL }}
+          ROL_API_KEY: ${{ secrets.ROL_API_KEY }}
+          ROL_PROJECT: ${{ secrets.ROL_PROJECT }}
+          ROL_OVPN_CONF_BASE64: ${{ secrets.ROL_OVPN_CONF_BASE64 }}
+      - name: Get EVE
+        uses: actions/checkout@v2
+        with:
+          path: eve
+      - name: Setup packages
+        run: |
+          sudo apt update && sudo apt upgrade
+          sudo apt install -y curl golang net-tools qemu binfmt-support qemu-user-static qemu-utils qemu-system-x86 qemu-system-aarch64 openvpn jq
+      - name: Setup OVPN connection to RoL RPi4 demo
+        id: vpn
+        run: |
+          echo "$ROL_OVPN_CONF_BASE64" | base64 -d > ./config.ovpn
+          sudo openvpn --config ./config.ovpn --daemon
+          until ip -f inet addr show tun0; do sleep 5; ip a; done
+          echo ::set-output name=ip::$(ip -f inet addr show tun0 | sed -En -e 's/.*inet ([0-9.]+).*/\1/p')
+        env:
+          ROL_OVPN_CONF_BASE64: ${{ secrets.ROL_OVPN_CONF_BASE64 }}
+      - name: Prepare eden
+        run: |
+          if [ -f ${{ github.workspace }}/eve/tests/eden/eden-version ]; then
+            EDEN_VERSION=$(cat ${{ github.workspace }}/eve/tests/eden/eden-version)
+          else
+            EDEN_VERSION=lfedge/eden:30c8b3f
+          fi
+          docker run -v $PWD:/out $EDEN_VERSION cp -a /eden/. /out/
+          sudo chown -R $(whoami) .
+      - name: Setup eden
+        run: |
+          ./eden config add default --devmodel=general --arch=arm64
+          ./eden config set default --key eve.tag --value="snapshot"
+          ./eden config set default --key eve.hv --value ${{ matrix.hv }}
+          ./eden config set default --key adam.eve-ip --value ${{ steps.vpn.outputs.ip }}
+          ./eden config set default --key registry.ip --value ${{ steps.vpn.outputs.ip }}
+          ./eden config set default --key=eden.tests --value=${{ github.workspace }}/eve/tests/eden
+          ./eden setup -v debug --netboot=true
+          ./eden start
+      - name: Get Rack of Labs CLI
+        uses: actions/checkout@v2
+        with:
+          repository: insei/rolgo-cli
+          path: rolgo-cli
+      - name: Build Rack of Labs CLI
+        run: |
+          cd rolgo-cli && go build
```

Comment:
Can't we publish a pre-built archive with this thing? Even better, can't we have a wrapper for it in Eden itself somehow?

Reply:
Now RoL is a part of EDEN.